### PR TITLE
converting to an adaptive card and adding link

### PR DESCRIPTION
**Pull request by walteck**
## Description

Adding ability to display a link button through the notification in teams

## Context

Enables users to optionally provide a link and to display this in the notification. Also switched to using AdaptiveCards for the incoming webhook format.

## Type of changes

- [X] Refactoring (non-breaking change)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [X] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [X] I have followed the code style of the project
- [X] I have added tests to cover my changes
- [X] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [X] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: src/messagecard.ts [hunk 1]
```diff
@@ -9,38 +9,110 @@ export function buildMessageCard(
...
+      }
+    ]
+  }
+
+  const adaptiveCard = {
+    type: 'AdaptiveCard',
+    $schema: 'http://adaptivecards.io/schemas/adaptive-card.json',
+    version: '1.4',
+    body: [title, subtitle, message, facts],
+    actions
```
Comment (jacobgill1) [inline src/messagecard.ts:103]: Does this need a key?

File: src/messagecard.ts [hunk 2]
```diff
@@ -9,38 +9,110 @@ export function buildMessageCard(
...
-  const card = {
-    '@type': 'MessageCard',
-    '@context': 'https://schema.org/extensions',
-    $schema: 'https://adaptivecards.io/schemas/adaptive-card.json',
-    version: '1.0',
-    summary: messageTitle,
-    themeColor: messageColour,
-    title: messageTitle,
-    sections: [
+  let actions: Record<string, string>[] = []
```
Comment (jacobgill1) [inline src/messagecard.ts:15]: Can this be a const?

**Review by jacobgill1 — COMMENTED** [review]
Just 2 comments

File: src/messagecard.ts (hunk 1 above)
Comment (walteck) [reply src/messagecard.ts:103]: This is covered under: https://eslint.org/docs/latest/rules/object-shorthand

File: src/messagecard.ts (hunk 2 above)
Comment (walteck) [reply src/messagecard.ts:15]: The actions variable is either an empty list or it is the full element, hence not being a const.

File: src/messagecard.ts (hunk 1 above)
Comment (jacobgill1) [reply src/messagecard.ts:103]: Ahh makes sense!

File: src/messagecard.ts (hunk 2 above)
Comment (jacobgill1) [reply src/messagecard.ts:15]: I didn't clock the reassignment on line 17

**Pull request merged**